docs/e2e/mobile-e2e-guidelines.md

```diff
@@ -0,0 +1,550 @@
+};
+```
+
+Or in terms of creating a selector string object:
+
+```javascript
+export const AddCustomTokenViewSelectorsText = {
+  TOKEN_SYMBOL: messages.token.token_symbol,
+  IMPORT_BUTTON: messages.add_asset.tokens.add_token,
+};
+```
+
+❌ Bad:
+
+```javascript
+const DELETE_WALLET_INPUT_BOX_ID = 'delete-wallet-input-box';
+```
+
+## Assertions
+
+To ensure consistency and avoid redundancy in our test scripts, we've implemented an Assertion class. Whenever assertions are needed within tests, it's preferred to utilize methods provided by this class.
+
+✅ Good: Utilize assertion methods from the Assertion class to perform assertions in tests. For instance:
+
+```javascript
+await Assertions.checkIfToggleIsOn(SecurityAndPrivacy.metaMetricsToggle);
+```
+
+❌ Bad: Refrain from using built-in assertion methods directly within tests, as shown below:
+
+```javascript
+await expect(element(by.id(Login - button))).toHaveText(login - text);
+```
+
+_NOTE: Generally speaking, you don’t need to put an assertion before a test action. You can minimize the amount of assertions by using a test action as an implied assertion on the same element._
+
+## 🛠 Method Design
+
+✅ Good: Embrace the "Don't Repeat Yourself" (DRY) principle. Reuse existing page-object actions to prevent code duplication and maintain code integrity. Ensure each method has a singular purpose.
+
+❌ Bad: Cluttering methods with multiple tasks and duplicating code, like this:
+
+```javascript
+
+tapNoThanksButton() {
+  await OnboardingWizard.isVisible();
+  await TestHelpers.waitAndTap(NO_THANKS_BUTTON_ID);
+  await TestHelpers.isNoThanksButtonNotVisible();
+}
+```
+
+## 📝 Comments
+
+✅ Good: Use JSDoc syntax for adding documentation to clarify complex logic or provide context. Excessive comments can be counterproductive and may signal a need for code simplification.
+
+```javascript
+
+/**
+  * Get element by web ID.
+  *
+  * @param {string} webID - The web ID of the element to locate
+  * @return {Promise} Resolves to the located element
+  */
+ async getElementByWebID(webID) {
+   return web.element(by.web.id(webID));
+ }
+```
+
+❌ Bad: Over Commenting simple and self-explanatory code, like this:
+
+```javascript
+
+// This function taps the button
+tapButton() {
+  // ...
+```
+
+## Creating Page Objects:
+
+Page objects serve as the building blocks of our test suites, providing a clear and organized representation of the elements and interactions within our application.
+
+When creating page objects, follow these principles to ensure clarity, maintainability, and reusability.
+
+### 🔁 Consistency in Structure
+
+Define a clear structure for your page objects, organizing elements and actions in a logical manner. This makes it easier for other engineers to understand and navigate the code.
+
+✅ Good: Define a clear structure for your page objects, organizing elements and actions in a logical manner. This makes it easy to reuse elements and actions across multiple tests. For example:
```

Comment:
Does 'Define a clear structure for your page objects, organizing elements and actions in a logical manner.' intentionally duplicates first sentence in line 249?

Reply:
oh wonderful eye, this was not intentional. I have updated it.